### PR TITLE
Check in/72443/demo fail

### PR DESCRIPTION
**Pull request by brianseek**
## Summary

This is a spin off ticket from the group 4 (errors) unified e2e ticket. This addresses failing to an error (see staff) page if the demographics patch fails. This ticket also adds the e2e test.

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#72443

## Testing done

- Adds e2e test for failed demographics patch

## What areas of the site does it impact?

Unified day-of check-in

## Acceptance criteria
The user is taken to a See Staff page if the demographics PATCH request fails.

### CODE REVIEW

**Review by rmessina1010 — APPROVED** [review]
LGTM

**Review by wullaski — APPROVED** [review]
LGTM!

File: src/applications/check-in/hooks/useSendDemographicsFlags.jsx [hunk 1]
```diff
@@ -30,12 +31,14 @@ const useSendDemographicsFlags = () => {
           .patchDayOfDemographicsData(demographicsData)
           .then(resp => {
             if (resp.data.error || resp.data.errors) {
-              throw new Error();
+              setError(true);
             } else {
               setShouldSendDemographicsFlags(window, false);
```
Comment (lee-delarm6) [inline src/applications/check-in/hooks/useSendDemographicsFlags.jsx:36]: Do you want to `setError(false)` when in the else section of this if statement?

**Review by lee-delarm6 — APPROVED** [review]
Approved

**Pull request merged**